draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
+
+For example, the
+SHA-256 hash digest of the Disclosure `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0` would be
+`uutlBuYeMDyjLLTpf6Jxi7yNkEF35jdyWMn9U7b_RYY`.
+
+#### Decoy Digests {#decoy_digests}
+
+An Issuer MAY add additional hash digests to the SD-JWT that are not associated with any claim.  The purpose of such "decoy" digests is to make it more difficult for an attacker to see the original number of claims contained in the SD-JWT. It is RECOMMENDED to create the decoy digests by hashing over a cryptographically secure random number. The bytes of the hash digest MUST then be base64url-encoded as above. The same digest function as for the Disclosures MUST be used.
+
+For decoy digests, no Disclosure is sent to the Holder, i.e., the Holder will see hash digests that do not correspond to any Disclosure. See (#decoy_digests_privacy) for additional privacy considerations.
+
+To ensure readability and replicability, the examples in this specification do not contain decoy digests unless explicitly stated.
+
+#### Creating an SD-JWT {#creating_sd_jwt}
+
+An SD-JWT is a JWT that MUST be signed using the Issuer's private key.
+
+An SD-JWT MAY contain both selectively disclosable claims and non-selectively disclosable claims, i.e., claims that are always contained in the SD-JWT in plaintext and are always visible to a Verifier.
+
+Claims controlling the validity of the SD-JWT, such as `iss`, `exp`, or `nbf` are usually included in plaintext. End-User claims MAY be included as plaintext as well, e.g., if hiding the particular claims from the Verifier does not make sense in the intended use case. Ultimately, an Issuer decides which claims are selectively disclosable and which are not.
+
+Plaintext claims are included in the SD-JWT just as they would be in any other JWT.
+
+Selectively disclosable claims are omitted from the SD-JWT. Instead, the hash digests of the respective Disclosures and potentially decoy digests are contained as an array in a new claim, `_sd`.
+
+The `_sd` claim MUST be an array of strings, each string being a hash digests of a Disclosure or a decoy digest as described above.
+The array MAY be empty, although it is RECOMMENDED to omit the claim in this case to save space.
+
+The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.
+
+Issuers MUST NOT issue SD-JWTs where
+
+ * the key `_sd` is used for any other purpose than to contain the array of hash digests, or
+ * the key `_sd` is itself used inside a claim value (when the claim value is an object), or
+ * the same Disclosure appears more than once (in the same array or in different arrays).
```

Comment:
when would this be the case? same claim value? same salt? I though we are recommending against this?

Reply:
Since the same salt should not happen, it might be an error of the Issuer or intentional misbehavior. It is ruled out here and enforced by the Verifier.

Reply:
```suggestion
 * the same Disclosure value appears more than once (in the same array or in different arrays).
```